ambry-cloud/src/main/java/com.github.ambry.cloud/azure/AzureBlobDataAccessor.java

```diff
@@ -142,24 +160,105 @@ public boolean uploadIfNotExists(BlobId blobId, long inputLength, CloudBlobMetad
     }
   }
 
+  /**
+   * Upload a file to blob storage.
+   * @param containerName name of the container where blob is stored.
+   * @param fileName the blob filename.
+   * @param inputStream the input stream to use for upload.
+   * @throws BlobStorageException for any error on ABS side.
+   * @throws IOException for any error with supplied data stream.
+   */
+  public void uploadFile(String containerName, String fileName, InputStream inputStream)
+      throws BlobStorageException, IOException {
+    try {
+      BlobContainerClient containerClient = getContainer(containerName, true);
+      BlockBlobClient blobClient = containerClient.getBlobClient(fileName).getBlockBlobClient();
```

Comment:
Could the method `getBlockBlobClient(String containerName, String blobName, boolean autoCreateContainer)` be used here?